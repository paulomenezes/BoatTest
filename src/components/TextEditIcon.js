import React from 'React';
import { TextInput, View } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import styles from './TextEdit.styles';

const TextEditIcon = props => {
  const { input, meta, style, ...inputProps } = props;

  const validationStyles =
    meta.touched && !meta.active
      ? !meta.error ? styles.validField : styles.invalidField
      : styles.validField;

  return (
    <View>
      <View style={[styles.inputArea, styles.inputIconArea, validationStyles]}>
        <Icon name="map-marker" size={20} color="#020202" style={styles.icon} />
        <TextInput
          {...inputProps}
          style={styles.input}
          placeholderTextColor="#020202"
          onChangeText={input.onChange}
          onBlur={input.onBlur}
          onFocus={input.onFocus}
          value={input.value}
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  );
};

export default TextEditIcon;
