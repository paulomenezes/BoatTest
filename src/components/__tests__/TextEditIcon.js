import 'react-native';
import React from 'react';
import TextEditIcon from '../TextEditIcon.js';

import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

it('renders correctly', () => {
  const props = {
    meta: {
      touched: false,
      active: false,
      error: undefined
    },
    input: {}
  };
  const tree = renderer.create(<TextEditIcon {...props} />);

  expect(tree).toMatchSnapshot();
});

it('Gray border when not touched', () => {
  const props = {
    meta: {
      touched: false,
      active: true,
      error: undefined
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEditIcon {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#CFCFCF');
});

it('Gray border when is valid', () => {
  const props = {
    meta: {
      touched: true,
      active: false,
      error: undefined
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEditIcon {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#CFCFCF');
});

it('Gray border when invalid', () => {
  const props = {
    meta: {
      touched: true,
      active: false,
      error: true
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEditIcon {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#D75452');
});
