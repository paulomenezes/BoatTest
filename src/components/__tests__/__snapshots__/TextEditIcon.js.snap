// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#E6E6E6",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "borderColor": "#CFCFCF",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#020202",
            "fontSize": 20,
          },
          Object {
            "paddingLeft": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      onBlur={undefined}
      onChangeText={undefined}
      onFocus={undefined}
      placeholderTextColor="#020202"
      style={
        Object {
          "flex": 1,
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
        }
      }
      value={undefined}
    />
  </View>
</View>
`;
